<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>window</class>
 <widget class="QMainWindow" name="window">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>550</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>950</width>
    <height>550</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MQTT Explorer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Dashboard</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="17,0,10">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QTreeWidget" name="treeWidget">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="indentation">
               <number>10</number>
              </property>
              <property name="uniformRowHeights">
               <bool>false</bool>
              </property>
              <property name="itemsExpandable">
               <bool>true</bool>
              </property>
              <property name="allColumnsShowFocus">
               <bool>false</bool>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
              <property name="headerHidden">
               <bool>false</bool>
              </property>
              <property name="columnCount">
               <number>2</number>
              </property>
              <attribute name="headerVisible">
               <bool>true</bool>
              </attribute>
              <attribute name="headerCascadingSectionResizes">
               <bool>true</bool>
              </attribute>
              <attribute name="headerMinimumSectionSize">
               <number>250</number>
              </attribute>
              <attribute name="headerHighlightSections">
               <bool>false</bool>
              </attribute>
              <attribute name="headerShowSortIndicator" stdset="0">
               <bool>false</bool>
              </attribute>
              <attribute name="headerStretchLastSection">
               <bool>true</bool>
              </attribute>
              <column>
               <property name="text">
                <string>Topics</string>
               </property>
               <property name="font">
                <font>
                 <weight>50</weight>
                 <bold>false</bold>
                </font>
               </property>
               <property name="textAlignment">
                <set>AlignLeading|AlignVCenter</set>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Last Message</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="3,1,0">
              <item>
               <widget class="QLineEdit" name="subscribe_text">
                <property name="placeholderText">
                 <string>e.g. home/cofeemachine/1</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="subscribe_btn">
                <property name="text">
                 <string>Subscribe new topic</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="unsubscribe_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="palette">
                 <palette>
                  <active>
                   <colorrole role="Button">
                    <brush brushstyle="SolidPattern">
                     <color alpha="255">
                      <red>168</red>
                      <green>50</green>
                      <blue>50</blue>
                     </color>
                    </brush>
                   </colorrole>
                   <colorrole role="ButtonText">
                    <brush brushstyle="SolidPattern">
                     <color alpha="255">
                      <red>255</red>
                      <green>255</green>
                      <blue>255</blue>
                     </color>
                    </brush>
                   </colorrole>
                  </active>
                  <inactive>
                   <colorrole role="Button">
                    <brush brushstyle="SolidPattern">
                     <color alpha="255">
                      <red>239</red>
                      <green>239</green>
                      <blue>239</blue>
                     </color>
                    </brush>
                   </colorrole>
                   <colorrole role="ButtonText">
                    <brush brushstyle="SolidPattern">
                     <color alpha="255">
                      <red>0</red>
                      <green>0</green>
                      <blue>0</blue>
                     </color>
                    </brush>
                   </colorrole>
                  </inactive>
                  <disabled>
                   <colorrole role="Button">
                    <brush brushstyle="SolidPattern">
                     <color alpha="255">
                      <red>239</red>
                      <green>239</green>
                      <blue>239</blue>
                     </color>
                    </brush>
                   </colorrole>
                   <colorrole role="ButtonText">
                    <brush brushstyle="SolidPattern">
                     <color alpha="255">
                      <red>190</red>
                      <green>190</green>
                      <blue>190</blue>
                     </color>
                    </brush>
                   </colorrole>
                  </disabled>
                 </palette>
                </property>
                <property name="text">
                 <string>Unsubscribe</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2" stretch="20,2,1">
            <item>
             <widget class="QListWidget" name="listWidget">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="isWrapping" stdset="0">
               <bool>false</bool>
              </property>
              <property name="resizeMode">
               <enum>QListView::Fixed</enum>
              </property>
              <property name="viewMode">
               <enum>QListView::ListMode</enum>
              </property>
              <property name="uniformItemSizes">
               <bool>false</bool>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0">
              <item>
               <widget class="QTextEdit" name="textEdit">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>50</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="send_btn">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>50</width>
                  <height>50</height>
                 </size>
                </property>
                <property name="text">
                 <string>Send</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="attachFile_btn">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>50</width>
                <height>50</height>
               </size>
              </property>
              <property name="text">
               <string>Send Image</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>All messages</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QListWidget" name="listWidget_all"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Widgets</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4" columnstretch="10,4">
        <item row="0" column="0">
         <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,50">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="layout_w">
            <item>
             <widget class="QScrollArea" name="widgets_scrollArea">
              <property name="verticalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOn</enum>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAsNeeded</enum>
              </property>
              <property name="sizeAdjustPolicy">
               <enum>QAbstractScrollArea::AdjustIgnored</enum>
              </property>
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents_2">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>626</width>
                 <height>407</height>
                </rect>
               </property>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <layout class="QVBoxLayout" name="widgets_layout">
                  <property name="sizeConstraint">
                   <enum>QLayout::SetDefaultConstraint</enum>
                  </property>
                  <item>
                   <spacer name="verticalSpacer_2">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Choose widget</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="widget_combobox">
            <item>
             <property name="text">
              <string>Lights</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Thermostat</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Coffe Machine</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Termometer</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Camera</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Name</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="widget_name"/>
          </item>
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Connect to</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="widget_topic">
            <property name="placeholderText">
             <string>e.g. home/cofeemachine</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="addWidget_btn">
            <property name="text">
             <string>Add widget</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>950</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSnapshot"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSnapshot">
   <property name="text">
    <string>Snapshot</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
